pkg/vz/save_restore_arm64.go

```diff
@@ -0,0 +1,76 @@
+//go:build darwin && arm64 && !no_vz
+
+package vz
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/Code-Hex/vz/v3"
+	"github.com/sirupsen/logrus"
+)
+
+func saveVM(vm *vz.VirtualMachine, machineStatePath string) error {
+	if !vm.CanPause() {
+		return fmt.Errorf("can't pause the VZ machine")
+	}
+
+	// Remove the old machine state file if it exists,
+	// because saving the machine state will fail if the file already exists.
+	if err := os.Remove(machineStatePath); err != nil && !errors.Is(err, os.ErrNotExist) {
+		logrus.WithError(err).Errorf("Failed to remove the old VZ machine state file %q", machineStatePath)
+		return err
+	}
+
+	logrus.Info("Pausing VZ")
+	if err := vm.Pause(); err != nil {
+		return err
+	}
+
+	// If we can't stop the machine after pausing, saving the machine state will be useless.
+	// So we should check this before saving the machine state.
+	if !vm.CanStop() {
+		return fmt.Errorf("can't stop the VZ machine after pausing")
+	}
```

Comment:
If we fail here the machine is left in paused state. Can we resume it to return to normal operation?

Reply:
I have moved this part to where resume is needed after an error.